### DISCUSSION TITLE
"[application] can't be opened because Apple cannot check it for malicious software" errors on Cask-installed apps

### DISCUSSION THREAD

**Discussion by chriskilding** [Homebrew/discussions#4294] [Casks]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

When I install some macOS apps via `brew cask`, the installation succeeds but macOS prevents the app from opening with the following error:

> [application] can't be opened because Apple cannot check it for malicious software.
>
> This software needs to be updated. Contact the developer for more information.

Most recently I've seen this with the JetBrains Toolbox Cask (version 1.27.3.14493), but I have also seen it at random with other apps installed by Cask as well.

In the case of JetBrains Toolbox, I then tried directly downloading the very same version of the app from the JetBrains site and installing that. This worked without issue. This make me think there is *something* going on within Homebrew Cask that is breaking app notarisation... but only sometimes.

Because this only happens sometimes, I'm posting this as a question rather than a bug report. If you have seen this behaviour as well, I would welcome your comments.

### Relevant casks

jetbrains-toolbox

**Comment by dkam** [comment]
I just discovered the same thing with `rar` and `unrar`, installed with homebrew. Reinstalling didn't help. MacOSX 13.4.

**Comment by fkurushin** [comment]
@dkam @chriskilding go to settings -> privacy and security than press allow do it for both `rar` and `unrar` and than run this commands in the terminal and press open

**Reply by andrewdai5** [reply]
worked for me.

**Comment by jaraco** [comment]
Another workaround is to use the Show in Finder option (from the error), and then explicitly "Open" the executable. Using that workflow, macOS will prompt if you really want to open this untrusted software. Saying yes will allow it to be run, whereafter it will work in the terminal.

**Comment by jaraco** [comment]
A couple of workarounds have been proposed, but it would be nice if there were a more durable solution built into Homebrew, such as to install the programs in such a way as to be trusted. Are they untrusted because they're unsigned? Could they be signed?

**Reply by p-linnane** [reply]
They need to be signed by their authors. We no longer allowed unsigned applications to be added to Homebrew/cask, and will begin removing existing unsigned apps at some point.

**Reply by astrid-dev** [reply]
I dont know how I feel about this, on one hand is great that you are strict by the sign of the package, but on the hand I'm unable to run the application I need and probably it will not be updated.

for anyone curious, is "font forge" and the last update is from January first 2023

**Reply by p-linnane** [reply]
Apple has been making it harder to launch unsigned software in recent macOS releases. We are simply following along since unsigned software throws lots of warnings and causes user confusion.

You can host your own tap with `fontforge` in it, or I can add it to my tap of unsigned applications for you https://github.com/p-linnane/homebrew-tap.
